internal/runbits/dependencies/changesummary.go

```diff
@@ -22,126 +22,70 @@ const showUpdatedPackages = true
-	directDependencies := make([]artifact.ArtifactID, 0, len(dependencies))
-	uniqueDependencies := make(map[artifact.ArtifactID]bool)
-	for artifactId, indirectDependencies := range dependencies {
-		directDependencies = append(directDependencies, artifactId)
-		uniqueDependencies[artifactId] = true
-		for _, depId := range indirectDependencies {
-			uniqueDependencies[depId] = true
-		}
-	}
-	sort.SliceStable(directDependencies, func(i, j int) bool {
-		return artifacts[directDependencies[i]].Name < artifacts[directDependencies[j]].Name
-	})
-	hasAdditionalIndirectDependencies := len(directDependencies) < len(uniqueDependencies)
+	dependencies = sliceutils.UniqueByProperty(dependencies, func(i *buildplan.Ingredient) any { return i.IngredientID })
+	directDependencies = sliceutils.UniqueByProperty(directDependencies, func(i *buildplan.Ingredient) any { return i.IngredientID })
+	numIndirect := len(dependencies) - len(directDependencies)
 
-	logging.Debug("%s has %d direct dependencies and %d total, unique dependencies", added.Name, len(directDependencies), len(uniqueDependencies))
+	logging.Debug("package %s has %d direct dependencies and %d total, unique dependencies",
+		strings.Join(addedString, ", "), len(directDependencies), numIndirect)
 	if len(directDependencies) == 0 {
 		return
 	}
 
 	// Process the existing runtime requirements into something we can easily compare against.
-	oldRequirements := make(map[string]string)
-	for _, source := range filter {
-		oldRequirements[fmt.Sprintf("%s/%s", source.Namespace, source.Name)] = *source.Version
-	}
+	oldRequirements := alreadyInstalled.Ingredients().ToIDMap()
 
 	localeKey := "additional_dependencies"
-	if hasAdditionalIndirectDependencies {
+	if numIndirect > 0 {
 		localeKey = "additional_total_dependencies"
 	}
-	version := ""
-	if added.Version != nil {
-		version = *added.Version
-	}
 	out.Notice(locale.Tr(localeKey,
-		added.Name, version, strconv.Itoa(len(directDependencies)), strconv.Itoa(len(uniqueDependencies))))
+		strings.Join(addedLocale, ", "), strconv.Itoa(len(dependencies)), strconv.Itoa(numIndirect)))
 
 	// A direct dependency list item is of the form:
 	//   ├─ name@version (X dependencies)
 	// or
 	//   └─ name@oldVersion → name@newVersion (Updated)
 	// depending on whether or not it has subdependencies, and whether or not showUpdatedPackages is
 	// `true`.
-	for i, artifactId := range directDependencies {
+	for i, ingredient := range directDependencies {
 		prefix := "├─"
 		if i == len(directDependencies)-1 {
 			prefix = "└─"
 		}
-		dep := artifacts[artifactId]
-
-		version := ""
-		if dep.Version != nil {
-			version = *dep.Version
-		}
 
+		ingredientDeps := ingredient.RuntimeDependencies(true)
 		subdependencies := ""
-		if numSubs := len(dependencies[dep.ArtifactID]); numSubs > 0 && hasAdditionalIndirectDependencies {
+		if numSubs := len(ingredientDeps); numSubs > 0 {
```

Comment:
There have been a lot of different changes in this file. Did you do an eyeball test for installing various packages? `state install requests`, `state install scipy` come to mind.